test/capital-project/capital-project.e2e-spec.ts

```diff
@@ -145,4 +148,87 @@ describe("Capital Projects", () => {
       expect(response.body.error).toBe(HttpName.INTERNAL_SEVER_ERROR);
     });
   });
+
+  describe("findCapitalCommitmentsByManagingCodeCapitalProjectId", () => {
+    it("should 200 and return an array of capital commitments", async () => {
+      const capitalProjectMock =
+        capitalProjectRepository.checkByManagingCodeCapitalProjectIdMocks[0];
+
+      const { managingCode, id: capitalProjectId } = capitalProjectMock;
+      const response = await request(app.getHttpServer())
+        .get(
+          `/capital-projects/${managingCode}/${capitalProjectId}/capital-commitments`,
+        )
+        .expect(200);
+
+      expect(() =>
+        findCapitalCommitmentsByManagingCodeCapitalProjectIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+    });
+
+    it("should 400 when finding by a too long managing code", async () => {
+      const tooLongManagingCode = "0725";
+      const capitalProjectId = "JIRO";
+
+      const response = await request(app.getHttpServer())
+        .get(
+          `/capital-projects/${tooLongManagingCode}/${capitalProjectId}/capital-commitments`,
+        )
+        .expect(400);
+
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 when finding by a lettered managing code", async () => {
+      const letteredManagingCode = "FUJI";
+      const capitalProjectId = "JIRO";
+
+      const response = await request(app.getHttpServer())
+        .get(
+          `/capital-projects/${letteredManagingCode}/${capitalProjectId}/capital-commitments`,
+        )
+        .expect(400);
+
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 404 when finding by missing managing code and capital project id", async () => {
+      const letteredManagingCode = "567";
```

Comment:
Nit: The variable name describes it as "too long" but the actual "flaw" is that it's missing